test/Base_Tests/src/Network/URI_Spec.enso

```diff
@@ -338,6 +340,24 @@ add_specs suite_builder =
                     r1.should_fail_with Illegal_Argument
                     r1.catch.to_display_text . should_contain "Secrets are not allowed in HTTP connections, use HTTPS instead."
 
+        with_secret ~action =
+            secret = Enso_Secret.create "my_test_secret-clear-cache-1-"+Random.uuid "My Value"
+            secret.should_succeed
+            Panic.with_finalizer secret.delete (action secret)
+
+        group_builder.specify "EnsoSecreetReader cache should be cleared when a reload is detected" pending=cloud_setup.pending <| Test.with_retries <|
+            cloud_setup.with_prepared_environment <|
+                with_secret secret1-> with_secret secret2->
+                    base_uri = URI.from 'https://httpbin.org/bytes/50'
+                    uri1 = base_uri . add_query_argument "arg1" secret1
+                    uri2 = base_uri . add_query_argument "arg1" secret2
+                    uri1.fetch . should_succeed
+                    uri2.fetch . should_succeed
+                    (EnsoSecretHelper.getEnsoSecretReaderCacheSize >= 2) . should_be_true
+                    EnsoSecretHelper.simulateEnsoSecretReaderReload
+                    uri1.fetch . should_succeed
+                    (EnsoSecretHelper.getEnsoSecretReaderCacheSize == 1) . should_be_true
```

Comment:
This is a good test because its core relies on `HTTP.fetch` method which is a real world method, and only uses test-specific methods to see into the internals to verify the behaviour.